WordPress/src/main/java/org/wordpress/android/util/CrashLogging.kt

```diff
@@ -0,0 +1,143 @@
+import org.wordpress.android.util.AppLog.T
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val EVENT_BUS_MODULE = "org.greenrobot.eventbus"
+private const val EVENT_BUS_EXCEPTION = "EventBusException"
+private const val EVENT_BUS_INVOKING_SUBSCRIBER_FAILED_ERROR = "Invoking subscriber failed"
+
+@Singleton
+class CrashLogging @Inject constructor(
+    private val accountStore: AccountStore
+) {
+    fun start(context: Context) {
+        SentryAndroid.init(context) { options ->
+            options.dsn = BuildConfig.SENTRY_DSN
+            options.cacheDirPath = context.cacheDir.absolutePath
+            options.isEnableSessionTracking = true // Release Health tracking
+            options.setBeforeSend { event, _ ->
+
+                if (!this.shouldSendEvents(context)) return@setBeforeSend null
+
+                if (event.exceptions.size > 1) {
+                    event.exceptions.lastOrNull()?.let { lastException ->
+                        // Remove the "Invoking subscriber failed" exception so that the main error will show up
+                        // in Sentry. This error only means that an exception occurred during an EventBus event and
+                        // it's not particularly useful for debugging.
+                        if (lastException.module == EVENT_BUS_MODULE &&
+                                lastException.type == EVENT_BUS_EXCEPTION &&
+                                lastException.value == EVENT_BUS_INVOKING_SUBSCRIBER_FAILED_ERROR) {
+                            event.exceptions.remove(lastException)
+                        }
+                    }
+                }
+                event
+            }
+        }
+
+        Sentry.setTag("version", BuildConfig.VERSION_NAME)
+
+        this.accountStore.account.apply {
+            val user = User()
+            user.id = this.userId.toString()
+            user.username = this.userName
+            user.email = this.email
+
+            Sentry.setUser(user)
+        }
+    }
+
+    /**
+     * Logs a message as a Sentry breadcrumb
+     *
+     * This doesn't generate a Sentry event – it just records it as context for any future crashes. To
+     * generate a Sentry event, use [report].
+     *
+     * @param[message] The message to log
+     * @param[tag] An optional [AppLog] tag
+     */
+    @JvmOverloads
+    fun log(message: String, tag: AppLog.T? = null) = tag?.let {
+        Sentry.addBreadcrumb(message, tag.toString())
+    } ?: run {
+        Sentry.addBreadcrumb(message)
+    }
+
+    /**
+     * Logs as an exception as a Sentry Breadcrumb
+     *
+     * This doesn't generate a Sentry event – it just records it as context for any future crashes. To
+     * generate a Sentry event, use [reportException].
+     *
+     * @param[message] The message to log
+     * @param[tag] An optional [AppLog] tag
+     */
+    @JvmOverloads
+    fun logException(tr: Throwable, tag: AppLog.T? = null) = this.log(tr.toString(), tag)
```

Comment:
Can we use `throwable` instead of `tr` for the parameter name? Named arguments are pretty common in Kotlin and I don't think `tr` immediately means `throwable`. This applies to `reportException` method as well.

Reply:
For sure – fixed in a34c5b762656e4ed3d17f5e0823dd69a7b1f187b